src/SFA.DAS.Campaign.UnitTests/Web/Controllers/EmployerInform/FundingAndApprenticeship/WhenFundingPostCalled.cs

```diff
@@ -1,30 +1,54 @@
+using System.Threading;
+using System.Threading.Tasks;
+using AutoFixture.NUnit3;
 using FluentAssertions;
 using MediatR;
 using Microsoft.AspNetCore.Mvc;
 using Moq;
 using NUnit.Framework;
+using SFA.DAS.Campaign.Domain.Content;
+using SFA.DAS.Campaign.Infrastructure.Api.Queries;
 using SFA.DAS.Campaign.Web.Controllers.EmployerInform;
 using SFA.DAS.Campaign.Web.Helpers;
+using SFA.DAS.Testing.AutoFixture;
 
 namespace SFA.DAS.Campaign.UnitTests.Web.Controllers.EmployerInform.FundingAndApprenticeship
 {
-    [TestFixture]
     public class WhenFundingPostCalled
     {
-        [Test]
-        public void Then_Vm_Is_Stored_In_Session()
+        [Test, RecursiveMoqAutoData]
+        public async Task Then_Vm_Is_Stored_In_Session(
+            LevyOptionViewModel model,
+            GetMenuQueryResult<Menu> mediatorResult,
+            [Frozen] Mock<IMediator> mediator,
+            [Frozen] Mock<ISessionService> sessionService,
+            [Greedy] FundingAnApprenticeshipController controller)
         {
-            var sessionService = new Mock<ISessionService>();
-            var mediator = new Mock<IMediator>();
-
-            var controller = new FundingAnApprenticeshipController(sessionService.Object, mediator.Object);
-
-            var levyOptionViewModel = new LevyOptionViewModel() {LevyStatus = LevyStatus.Levy};
-            controller.Index(levyOptionViewModel);
+            //Act
+            await controller.Index(model);
             
-            sessionService.Verify(ss => ss.Set(sessionService.Object.LevyOptionViewModelKey, levyOptionViewModel), Times.Once());
+            //Assert
+            sessionService.Verify(ss => ss.Set(sessionService.Object.LevyOptionViewModelKey, model), Times.Once());
+            mediator.Verify(x=>x.Send(It.IsAny<GetMenuQuery>(), CancellationToken.None), Times.Never);
         }
 
-        
+        [Test, RecursiveMoqAutoData]
+        public async Task Then_If_There_Is_An_Error_The_Menu_Is_Build(
+            LevyOptionViewModel model,
+            GetMenuQueryResult<Menu> mediatorResult,
+            [Frozen] Mock<IMediator> mediator,
+            [Greedy] FundingAnApprenticeshipController controller)
+        {
+            //Arrange
+            mediator.Setup(x => x.Send(It.IsAny<GetMenuQuery>(), CancellationToken.None))
+                .ReturnsAsync(mediatorResult);
+            controller.ModelState.AddModelError("TestError","Error");
+           
+            //Act
+            await controller.Index(model);
+            
+            //Assert
+            mediator.Verify(x=>x.Send(It.IsAny<GetMenuQuery>(), CancellationToken.None), Times.Once);
```

Comment:
Shouldn't the assertion be that the Menu property of the VM is set to the query result or similar?

Super pedantic, but should the test name end _Built_? or _Added_?

Reply:
Have updated the test name and added assertion to check menu on viewmodel